libr/include/r_types.h

```diff
@@ -269,10 +269,16 @@ static inline void *r_new_copy(int size, void *data) {
 	}
 	return a;
 }
+// Return 0 for array type, build error for other types
+#ifdef __GNUC__
+#define R_MUST_BE_ARRAY(a) (sizeof(struct { int:(-!!(__builtin_types_compatible_p(typeof(a), typeof(&(a)[0])))); }))
+#else
+#define R_MUST_BE_ARRAY(a) 0
```

Comment:
Not much reliable outside gnuland. Probably better to not rely on this feature

Reply:
I do agree as it is right now it's not a feature we should rely on. Maybe we should change the name, otherwise people may think that by using it you can be sure the argument is an array.

Maybe `R_MUST_BE_ARRAY` -> `R_MAY_BE_ARRAY` or SHOULD...